core/data/src/commonMain/kotlin/io/github/droidkaigi/confsched2023/data/stamps/StampDataStore.kt

```diff
@@ -0,0 +1,42 @@
+package io.github.droidkaigi.confsched2023.data.stamps
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.core.IOException
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.stringPreferencesKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+
+class StampDataStore(
+    private val dataStore: DataStore<Preferences>,
+) {
+    internal suspend fun save(
+        isDisplayed: Boolean,
+    ) {
+        dataStore.edit { preferences ->
+            preferences[DATA_STORE_STAMPS_KEY] = isDisplayed.toString()
+        }
+    }
+
+    public fun isDisplayedStream(): Flow<Boolean?> {
+        return dataStore.data
+            .catch { exception ->
+                if (exception is IOException) {
+                    emit(emptyPreferences())
+                } else {
+                    throw exception
+                }
+            }
+            .map { preferences: Preferences ->
+                preferences[DATA_STORE_STAMPS_KEY]?.toBoolean()
```

Comment:
@takahirom @momomomo111 
I have a question.
I assumed that when the app is installed and the preferences[DATA_STORE_STAMPS_KEY] is accessed for the first time, null is returned.
However, in reality, true is always returned from the first access after installation.
I am investigating this part of the problem, but I am unable to identify which part of the code is returning true.
Any help would be appreciated.:bow: